core/src/main/java/org/ehcache/core/spi/service/LocalPersistenceService.java

```diff
@@ -17,22 +17,64 @@
 package org.ehcache.core.spi.service;
 
 import org.ehcache.CachePersistenceException;
-import org.ehcache.spi.persistence.PersistableResourceService;
+import org.ehcache.spi.service.MaintainableService;
 
 /**
- * Service to provide persistence context to caches requiring it.
- * <P>
- * Will be used by caches with a disk store, whether or not the data should survive a program restart.
- * <P/>
+ * Service that provides unique local persistence spaces (and safe sub spaces) to any service that requires it
+ * under the local root directory.
  */
-public interface LocalPersistenceService extends PersistableResourceService {
+public interface LocalPersistenceService extends MaintainableService {
 
   /**
-   * Creates a new persistence context within the given space.
+   * Prepare a logical safe directory space for the owner and return an identifying space Id.
    *
-   * @param identifier space to create within
-   * @param name name of the context to create
-   * @return a {@link FileBasedPersistenceContext}
+   * @param owner Service owner that owns the safe space.
+   * @param name Identifying name for the space.
+   *
+   * @return Opaque Identifier that can be used to identify the safe space.
+   */
+  SafeSpaceIdentifier createSafeSpaceIdentifier(String owner, String name);
+
+  /**
+   * Create the safe spaces in actual physical space for the owner.
+   *
+   * @param safeSpaceId prepared logical space on which the physical space
+   */
+  void createSafeSpace(SafeSpaceIdentifier safeSpaceId) throws CachePersistenceException;
+
+  /**
+   * Create a physical persistent context (directory) within the safe space.
+   *
+   * @param safeSpaceId Safe space identifier
+   * @return persistent context
    */
-  FileBasedPersistenceContext createPersistenceContextWithin(PersistenceSpaceIdentifier<?> identifier, String name) throws CachePersistenceException;
-}
+  FileBasedPersistenceContext createContextWithin(SafeSpaceIdentifier safeSpaceId, String name) throws CachePersistenceException;
+
+  /**
+   * Destroy the safe space.
+   *
+   * @param safeSpaceId Safe space identifier.
+   * @param verbose Log more information.
+   */
+  void destroySafeSpace(SafeSpaceIdentifier safeSpaceId, boolean verbose);
+
+
+  /**
+   * Destroy all safe spaces provided to this owner.
+   *
+   * @param owner owner of safe spaces.
+   */
+  void destroyAll(String owner);
+
+  /**
+   * Destroy safe space given owner and identifier.
+   *
+   * @param owner owner of safe spaces
+   * @param name name of safe spaces
+   * @param verbose log more
+   */
+  void destroySafeSpace(String owner, String name, boolean verbose);
```

Comment:
Given that the `SafeSpaceIdentifier` can be obtained without any file based action, this makes little sense as one can always invoke the overloaded method.